app/scripts/lib/app-start.js

```diff
@@ -243,15 +247,48 @@ function (
       this._metrics.init();
     },
 
-    initializeIframeChannel: function () {
-      if (this._isIframe()) {
-        this._iframeChannel = new IframeChannel();
-        this._iframeChannel.initialize({
-          window: this._window,
-          origin: this._relier.get('origin'),
-          metrics: this._metrics
-        });
+    _getExpectedIframeOrigin: function () {
+      if (! this._isInAnIframe()) {
+        return null;
       }
+      if (this._isFxDesktop()) {
+        // If in an iframe for sync, the origin is
+        // checked against a pre-defined origin sent
+        // from the server.
+        return this._config.firstRunOrigin;
```

Comment:
I'm not convinced this is the best way to do this because it limits us to a single origin, but I figure we can iterate if this is needed more generically.